examples/moving_least_squares/moving_least_squares.cpp

```diff
@@ -0,0 +1,335 @@
+  source_points_host(8) = ArborX::Point {  1., -1.,  2. };
+  source_points_host(9) = ArborX::Point {  0.,  0., -1. };
+  Kokkos::deep_copy(source_points, source_points_host);
+
+  // Generate target points
+  target_points_host(0) = ArborX::Point {  0.,  0.,  0. };
+  target_points_host(1) = ArborX::Point {  .5,  .5,  0. };
+  target_points_host(2) = ArborX::Point { -.5,  .5,  1. };
+  target_points_host(3) = ArborX::Point {  0., -.5, 1.5 };
+  Kokkos::deep_copy(target_points, target_points_host);
+
+  // Arrange source points as tree
+  auto source_tree =
+    ArborX::BVH<MemorySpace>(ExecutionSpace{}, source_points);
+  
+  // Create the queries
+  // For each target point we query the closest source points
+  auto queries = Kokkos::View<ArborX::Nearest<ArborX::Point>*, MemorySpace>(
+    "queries", target_points_num);
+  Kokkos::parallel_for(
+    "make_queries",
+    Kokkos::RangePolicy<ExecutionSpace>(0, target_points_num),
+    KOKKOS_LAMBDA (const int i) {
+      queries(i) = ArborX::nearest(target_points(i), num_neighbors);
+  });
+
+  // Perform the query
+  auto indices = Kokkos::View<int *, MemorySpace>("indices", 0);
+  auto offsets = Kokkos::View<int *, MemorySpace>("offsets", 0);
+  source_tree.query(ExecutionSpace{}, queries, indices, offsets);
+
+  // Now that we have the neighbors, we recompute their position using
+  // their target point as the origin.
+  // This is used as an optimisation later in the algorithm
+  auto tr_source_points = Kokkos::View<ArborX::Point**, MemorySpace>(
+    "tr_source_points", target_points_num, num_neighbors);
+  Kokkos::parallel_for(
+    "transform_source_points",
+    Kokkos::RangePolicy<ExecutionSpace>(0, target_points_num),
+    KOKKOS_LAMBDA (const int i) {
+      for (int j = offsets(i); j < offsets(i+1); j++) {
+        tr_source_points(i, j - offsets(i)) = ArborX::Point {
+          source_points(indices(j))[0] - target_points(i)[0],
+          source_points(indices(j))[1] - target_points(i)[1],
+          source_points(indices(j))[2] - target_points(i)[2],
+        };
+      }
+  });
+
+  // Compute the radii for the weight (phi) vector
+  auto radii = Kokkos::View<double*, MemorySpace>(
+    "radii", target_points_num);
+  constexpr double epsilon = std::numeric_limits<double>::epsilon();
+  Kokkos::parallel_for(
+    "radii_computation",
+    Kokkos::RangePolicy<ExecutionSpace>(0, target_points_num),
+    KOKKOS_LAMBDA (const int i) {
+      double radius = 10. * epsilon;
+
+      for (int j = 0; j < num_neighbors; j++) {
+        double norm = ArborX::Details::distance(
+          tr_source_points(i, j),
+          ArborX::Point{0., 0., 0.});
+        radius = (radius < norm) ? norm : radius;
+      }
+
+      radii(i) = 1.1 * radius;
+  });
+
+  // Compute the weight (phi) vector
+  auto phi = Kokkos::View<double**, MemorySpace>(
+    "phi", target_points_num, num_neighbors);
+  Kokkos::parallel_for(
+    "phi_computation",
```

Comment:
We also try to prefix the kernels to indicate the origin, like `Example::phi_computation`.